src/Materials/PBR/pbrBaseMaterial.ts

```diff
@@ -248,85 +253,33 @@ export class PBRMaterialDefines extends MaterialDefines
 
     public SPECULARAA = false;
 
-    public CLEARCOAT = false;
-    public CLEARCOAT_DEFAULTIOR = false;
-    public CLEARCOAT_TEXTURE = false;
-    public CLEARCOAT_TEXTURE_ROUGHNESS = false;
-    public CLEARCOAT_TEXTUREDIRECTUV = 0;
-    public CLEARCOAT_TEXTURE_ROUGHNESSDIRECTUV = 0;
-    public CLEARCOAT_USE_ROUGHNESS_FROM_MAINTEXTURE = false;
-    public CLEARCOAT_TEXTURE_ROUGHNESS_IDENTICAL = false;
-    public CLEARCOAT_BUMP = false;
-    public CLEARCOAT_BUMPDIRECTUV = 0;
-    public CLEARCOAT_REMAP_F0 = true;
-    public CLEARCOAT_TINT = false;
-    public CLEARCOAT_TINT_TEXTURE = false;
-    public CLEARCOAT_TINT_GAMMATEXTURE = false;
-    public CLEARCOAT_TINT_TEXTUREDIRECTUV = 0;
-
-    public ANISOTROPIC = false;
-    public ANISOTROPIC_TEXTURE = false;
-    public ANISOTROPIC_TEXTUREDIRECTUV = 0;
-
     public BRDF_V_HEIGHT_CORRELATED = false;
     public MS_BRDF_ENERGY_CONSERVATION = false;
     public SPECULAR_GLOSSINESS_ENERGY_CONSERVATION = false;
 
-    public SHEEN = false;
-    public SHEEN_TEXTURE = false;
-    public SHEEN_GAMMATEXTURE = false;
-    public SHEEN_TEXTURE_ROUGHNESS = false;
-    public SHEEN_TEXTUREDIRECTUV = 0;
-    public SHEEN_TEXTURE_ROUGHNESSDIRECTUV = 0;
-    public SHEEN_LINKWITHALBEDO = false;
-    public SHEEN_ROUGHNESS = false;
-    public SHEEN_ALBEDOSCALING = false;
-    public SHEEN_USE_ROUGHNESS_FROM_MAINTEXTURE = false;
-    public SHEEN_TEXTURE_ROUGHNESS_IDENTICAL = false;
-
-    public SUBSURFACE = false;
-
-    public SS_REFRACTION = false;
-    public SS_REFRACTION_USE_INTENSITY_FROM_TEXTURE = false;
-    public SS_TRANSLUCENCY = false;
-    public SS_TRANSLUCENCY_USE_INTENSITY_FROM_TEXTURE = false;
-    public SS_SCATTERING = false;
-
-    public SS_THICKNESSANDMASK_TEXTURE = false;
-    public SS_THICKNESSANDMASK_TEXTUREDIRECTUV = 0;
-    public SS_HAS_THICKNESS = false;
-    public SS_REFRACTIONINTENSITY_TEXTURE = false;
-    public SS_REFRACTIONINTENSITY_TEXTUREDIRECTUV = 0;
-    public SS_TRANSLUCENCYINTENSITY_TEXTURE = false;
-    public SS_TRANSLUCENCYINTENSITY_TEXTUREDIRECTUV = 0;
-
-    public SS_REFRACTIONMAP_3D = false;
-    public SS_REFRACTIONMAP_OPPOSITEZ = false;
-    public SS_LODINREFRACTIONALPHA = false;
-    public SS_GAMMAREFRACTION = false;
-    public SS_RGBDREFRACTION = false;
-    public SS_LINEARSPECULARREFRACTION = false;
-    public SS_LINKREFRACTIONTOTRANSPARENCY = false;
-    public SS_ALBEDOFORREFRACTIONTINT = false;
-    public SS_ALBEDOFORTRANSLUCENCYTINT = false;
-    public SS_USE_LOCAL_REFRACTIONMAP_CUBIC = false;
-    public SS_USE_THICKNESS_AS_DEPTH = false;
-
-    public SS_MASK_FROM_THICKNESS_TEXTURE = false;
-    public SS_USE_GLTF_TEXTURES = false;
-
     public UNLIT = false;
 
     public DEBUGMODE = 0;
 
+    protected _keysFromPlugins?: string[];
+
     /**
      * Initializes the PBR Material defines.
+     * @param keysFromPlugins The plugin keys
      */
-    constructor() {
+    constructor(keysFromPlugins?: string[]) {

```

Comment:
I am not sure this signature can work as it does not take the type in account e.g. a number vs a boolean as well as the default values.

Reply:
Maybe:
```
keysFromPlugins: Array<{ name: string, type: string, default: any }>
```
If we agree we should move this to the `MaterialDefines` class instead, replace `keysFromPlugins` by `externalProperties` (better than `externalKeys` as we provide the full description of additional properties).

Reply:
happy with this :-)